src/bentoml/_internal/utils/__init__.py

```diff
@@ -303,6 +304,24 @@ def resolve_user_filepath(filepath: str, ctx: t.Optional[str]) -> str:
     raise FileNotFoundError(f"file {filepath} not found")
 
 
+def find_nested_requirements_txts(
+    base_requirements_txt: Path,
+) -> t.Tuple[Path, t.Set[Path]]:
+    """Recursively search the provided requirements.txt file to find nested
+    requirements files specified with `-r`.
+    """
+    sub_requirements_txt_files: t.Set[Path] = set()
```

Comment:
```suggestion
    sub_requirements_txt_files: set[Path] = set()
```

Reply:
That's only valid syntax in Python 3.9+ isn't it?

Reply:
we are using `from __future__` import